downloader/src/main/kotlin/Main.kt

```diff
@@ -293,7 +293,8 @@ object Main {
      */
     fun download(target: Package, outputDirectory: File): DownloadResult {
         // TODO: add namespace to path
-        val targetDir = File(outputDirectory, "${target.normalizedName}/${target.id.version}").apply { safeMkdirs() }
+        val version = if (target.id.version.isBlank()) "unknown" else target.id.version
```

Comment:
Maybe nicer:

    val version = target.id.version.takeIf { it.isNotBlank } ?: "unknown"

Reply:
Or:

    val version = target.id.version.takeUnless { it.isBlank() } ?: "unknown"

:wink:

Reply:
Done.